keras_hub/src/models/yolo_v8/yolo_v8_detector_test.py

```diff
@@ -0,0 +1,337 @@
+
+        yolo.compile(
+            optimizer="adam",
+            classification_loss="auto",
+            box_loss="auto",
+        )
+        xs, ys = _create_bounding_box_dataset(bounding_box_format)
+        # Make all bounding_boxes invalid and filter out them
+        ys["classes"] = -np.ones_like(ys["classes"])
+
+        yolo.fit(x=xs, y=ys, epochs=1)
+
+    def test_trainable_weight_count(self):
+        yolo = keras_hub.models.YOLOV8ObjectDetector(
+            num_classes=2,
+            fpn_depth=1,
+            bounding_box_format="xywh",
+            backbone=keras_hub.models.YOLOV8Backbone.from_preset(
+                "yolo_v8_s_backbone"
+            ),
+        )
+
+        self.assertEqual(len(yolo.trainable_weights), 195)
+
+    def test_bad_loss(self):
+        yolo = keras_hub.models.YOLOV8ObjectDetector(
+            num_classes=2,
+            fpn_depth=1,
+            bounding_box_format="xywh",
+            backbone=keras_hub.models.YOLOV8Backbone.from_preset(
+                "yolo_v8_xs_backbone"
+            ),
+        )
+
+        with self.assertRaisesRegex(
+            ValueError,
+            "Invalid box loss",
+        ):
+            yolo.compile(box_loss="bad_loss", classification_loss="auto")
+
+        with self.assertRaisesRegex(
+            ValueError,
+            "Invalid classification loss",
+        ):
+            yolo.compile(box_loss="auto", classification_loss="bad_loss")
+
+    @pytest.mark.large  # Saving is slow, so mark these large.
+    def test_saved_model(self):
+        model = keras_hub.models.YOLOV8ObjectDetector(
+            num_classes=20,
+            bounding_box_format="xywh",
+            fpn_depth=1,
+            backbone=keras_hub.models.YOLOV8Backbone.from_preset(
+                "yolo_v8_xs_backbone"
+            ),
+        )
+        xs, _ = _create_bounding_box_dataset("xywh")
+        model_output = model(xs)
+        save_path = os.path.join(
+            self.get_temp_dir(), "yolo_v8_xs_detector.keras"
+        )
+        model.save(save_path)
+        # TODO: Remove the need to pass the `custom_objects` parameter.
+        restored_model = keras.saving.load_model(
+            save_path,
+            custom_objects={
+                "YOLOV8ObjectDetector": keras_hub.models.YOLOV8ObjectDetector
+            },
+        )
+
+        # Check we got the real object back.
+        self.assertIsInstance(
+            restored_model, keras_hub.models.YOLOV8ObjectDetector
+        )
+
+        # Check that output matches.
+        restored_output = restored_model(xs)
+        self.assertAllClose(
+            ops.convert_to_numpy(model_output["boxes"]),
+            ops.convert_to_numpy(restored_output["boxes"]),
+        )
+        self.assertAllClose(
+            ops.convert_to_numpy(model_output["classes"]),
+            ops.convert_to_numpy(restored_output["classes"]),
+        )
+
+    # TODO(tirthasheshpatel): Support updating prediction decoder in Keras Core.
```

Comment:
is this todo relevant?

Reply:
This TODO is no longer relevant, so I have removed it.